nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -0,0 +1,484 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.slurm;
+  # configuration file can be generated by http://slurm.schedmd.com/configurator.html
+  configFile = pkgs.writeText "slurm.conf" 
+    ''
+      ${optionalString (cfg.controlMachine != null) ''controlMachine=${cfg.controlMachine}''}
+      ${optionalString (cfg.controlAddr != null) ''controlAddr=${cfg.controlAddr}''}
+      ${optionalString (cfg.authType != null) ''authType=${cfg.authType}''}
+      ${optionalString (cfg.cacheGroups != null) ''cacheGroups=${cfg.cacheGroups}''}
+      ${optionalString (cfg.cryptoType != null) ''cryptoType=${cfg.cryptoType}''}
+      ${optionalString (cfg.mpiDefault != null) ''mpiDefault=${cfg.mpiDefault}''}
```

Comment:
Well, ok. I have less tasks to do :)


On 2 March 2015 19:36:23 lethalman <notifications@github.com> wrote:

> > +{ config, lib, pkgs, ... }:
> > +
> > +with lib;
> > +
> > +let
> > +
> > +  cfg = config.services.slurm;
> > +  # configuration file can be generated by 
> http://slurm.schedmd.com/configurator.html
> > +  configFile = pkgs.writeText "slurm.conf"
> > +    ''
> > +      ${optionalString (cfg.controlMachine != null) 
> ''controlMachine=${cfg.controlMachine}''}
> > +      ${optionalString (cfg.controlAddr != null) 
> ''controlAddr=${cfg.controlAddr}''}
> > +      ${optionalString (cfg.authType != null) ''authType=${cfg.authType}''}
> > +      ${optionalString (cfg.cacheGroups != null) 
> ''cacheGroups=${cfg.cacheGroups}''}
> > +      ${optionalString (cfg.cryptoType != null) 
> ''cryptoType=${cfg.cryptoType}''}
> > +      ${optionalString (cfg.mpiDefault != null) 
> ''mpiDefault=${cfg.mpiDefault}''}
>
> Depends :) It may be easier to copy-paste a line of example config instead. 
> Options are useful if you're really in need of 
> merging/checking/manipulating them, otherwise they tend do be overkill. 
> E.g. if you had to run some scripts based on those partitions ecc., ok, or 
> if the config was more complex than a simple line.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/NixOS/nixpkgs/pull/6611/files#r25612669